fcrepo4/fcrepo4
# Make the default root ACL accessible from HTTP API.

Make the default root ACL of the repository available from HTTP API.

**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-2825

# What does this Pull Request do?
Retrieve the default root ACL from a used specified location if there's one exists, otherwise use the default one provided by Fedora and make it accessible from HTTP rest api. The default root ACL file is moved from the fcrepo-auth-webac module to fcrepo-http-api for resource retrieval.

# How should this be tested?
```
1. Test retrieval of the default ACL provided by Fedora.
Start Fedora with the following command:
$ mvn jetty:run -pl fcrepo-webapp

Verify that the Fedora provided default root ACL is retrieved:
$ curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/fcr:acl
HTTP/1.1 200 OK
Date: Fri, 17 Aug 2018 17:54:42 GMT
Set-Cookie: JSESSIONID=7q4rj8kgdxh5bqd7hlhj497q;Path=/
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Set-Cookie: rememberMe=deleteMe; Path=/; Max-Age=0; Expires=Thu, 16-Aug-2018 17:54:42 GMT
Content-Type: text/turtle;charset=utf-8
Content-Length: 1343
Server: Jetty(9.3.1.v20150714)

@prefix premis:  <http://www.loc.gov/premis/rdf/v1#> .
@prefix test:  <info:fedora/test/> .
@prefix memento:  <http://mementoweb.org/ns#> .
@prefix rdfs:  <http://www.w3.org/2000/01/rdf-schema#> .
@prefix webac:  <http://fedora.info/definitions/v4/webac#> .
@prefix acl:  <http://www.w3.org/ns/auth/acl#> .
@prefix ns001:  <http://purl.org/dc/terms/> .
@prefix xsi:  <http://www.w3.org/2001/XMLSchema-instance> .
@prefix xmlns:  <http://www.w3.org/2000/xmlns/> .
@prefix rdf:  <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix fedora:  <http://fedora.info/definitions/v4/repository#> .
@prefix xml:  <http://www.w3.org/XML/1998/namespace> .
@prefix ebucore:  <http://www.ebu.ch/metadata/ontologies/ebucore/ebucore#> .
@prefix ldp:  <http://www.w3.org/ns/ldp#> .
@prefix xs:  <http://www.w3.org/2001/XMLSchema> .
@prefix fedoraconfig:  <http://fedora.info/definitions/v4/config#> .
@prefix foaf:  <http://xmlns.com/foaf/0.1/> .
@prefix dc:  <http://purl.org/dc/elements/1.1/> .

<http://localhost:8080/rest/fcr:acl>
        acl:accessToClass  fedora:Resource ;
        acl:agentClass     foaf:Agent ;
        rdfs:comment       "By default, all non-Admin agents (foaf:Agent) are denied access (no acl:mode is specified) to all resources." ;
        rdfs:label         "Root Authorization" ;
        rdf:type           acl:Authorization .

2. Test retrieval of the default ACL from a user specified location.
Start Fedora with the following command:
$ mvn jetty:run -Dfcrepo.auth.webac.authorization=./fcrepo-http-api/target/test-classes/test-root-authorization.ttl  -pl fcrepo-webapp

Verify that the test root ACL is retrieved:
$ curl -i -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/fcr:acl
HTTP/1.1 200 OK
Date: Fri, 17 Aug 2018 17:56:30 GMT
Set-Cookie: JSESSIONID=1bsb2t55m0mlu1mpvw9iyzfpjf;Path=/
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Set-Cookie: rememberMe=deleteMe; Path=/; Max-Age=0; Expires=Thu, 16-Aug-2018 17:56:30 GMT
Content-Type: text/turtle;charset=utf-8
Content-Length: 1349
Server: Jetty(9.3.1.v20150714)

@prefix premis:  <http://www.loc.gov/premis/rdf/v1#> .
@prefix test:  <info:fedora/test/> .
@prefix memento:  <http://mementoweb.org/ns#> .
@prefix rdfs:  <http://www.w3.org/2000/01/rdf-schema#> .
@prefix webac:  <http://fedora.info/definitions/v4/webac#> .
@prefix acl:  <http://www.w3.org/ns/auth/acl#> .
@prefix ns001:  <http://purl.org/dc/terms/> .
@prefix xsi:  <http://www.w3.org/2001/XMLSchema-instance> .
@prefix xmlns:  <http://www.w3.org/2000/xmlns/> .
@prefix rdf:  <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix fedora:  <http://fedora.info/definitions/v4/repository#> .
@prefix xml:  <http://www.w3.org/XML/1998/namespace> .
@prefix ebucore:  <http://www.ebu.ch/metadata/ontologies/ebucore/ebucore#> .
@prefix ldp:  <http://www.w3.org/ns/ldp#> .
@prefix xs:  <http://www.w3.org/2001/XMLSchema> .
@prefix fedoraconfig:  <http://fedora.info/definitions/v4/config#> .
@prefix foaf:  <http://xmlns.com/foaf/0.1/> .
@prefix dc:  <http://purl.org/dc/elements/1.1/> .

<http://localhost:8080/rest/fcr:acl>
        acl:accessToClass  fedora:Resource ;
        acl:mode           acl:Read ;
        acl:agent          "testAdminUser" ;
        rdfs:comment       "Provide read access to all resources to the testAdminUser agent." ;
        rdfs:label         "(Test) Root Authorization" ;
        rdf:type           acl:Authorization .
```

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo4/committers
